gax-java/gax/src/main/java/com/google/api/gax/rpc/TransportChannelProvider.java

```diff
@@ -97,6 +97,13 @@ public interface TransportChannelProvider {
    */
   TransportChannelProvider withEndpoint(String endpoint);
 
+  /** Sets whether to use S2A when constructing a new {@link TransportChannel}. */
+  @BetaApi(
+      "The S2A feature is not stable yet and may change in the future. https://github.com/grpc/grpc-java/issues/11533.")
+  default TransportChannelProvider withUseS2A(boolean useS2A) {
+    throw new UnsupportedOperationException("S2A is not supported");
```

Comment:
I know we decided to throw exception last time we reviewed it, but we discovered that this could break downstream libraries' or customers' tests if they have a mock `TransportChannelProvider`. To prevent such breaking changes, can we change this to `return this;` instead? Then we don't have to override this method in other implementations (`LocalChannelProvider`, `InstantiatingHttpJsonChannelProvider` etc.) of this interface as well.

Reply:
Thanks for bringing this up! Changed the default behavior to `return this` and removed overriding of the method in all other implementations except gRPC.

https://github.com/googleapis/sdk-platform-java/pull/3548/commits/1703f51775763840028a7d3dc843cf0ccefbe68a